tests/topotests/ospf6-topo1-vrf/test_ospf6_topo1_vrf.py

```diff
@@ -0,0 +1,505 @@
+    )
+
+    cmds = [
+        "ip link add {0}-cust1 type vrf table 1001",
+        "ip link add loop1 type dummy",
+        "ip link set {0}-stubnet master {0}-cust1",
+    ]
+
+    cmds1 = [
+        "ip link set {0}-sw5 master {0}-cust1",
+    ]
+
+    cmds2 = [
+        "ip link set {0}-sw6 master {0}-cust1",
+    ]
+
+    # For all registered routers, load the zebra configuration file
+    for rname, router in router_list.iteritems():
+        # create VRF rx-cust1 and link rx-eth0 to rx-cust1
+        for cmd in cmds:
+            output = tgen.net[rname].cmd(cmd.format(rname))
+        if rname == "r1" or rname == "r2" or rname == "r3":
+            for cmd in cmds1:
+                output = tgen.net[rname].cmd(cmd.format(rname))
+        if rname == "r3" or rname == "r4":
+            for cmd in cmds2:
+                output = tgen.net[rname].cmd(cmd.format(rname))
+
+        output = tgen.net[rname].cmd("sysctl -n net.ipv4.tcp_l3mdev_accept")
+        logger.info(
+            "router {0}: existing tcp_l3mdev_accept was {1}".format(rname, output)
+        )
+
+        if l3mdev_accept:
+            output = tgen.net[rname].cmd(
+                "sysctl -w net.ipv4.tcp_l3mdev_accept={}".format(l3mdev_accept)
+            )
+
+   
+    for rname, router in router_list.iteritems():
+        router.load_config(
+            TopoRouter.RD_ZEBRA, os.path.join(CWD, "{}/zebra.conf".format(rname))
+        )
+        router.load_config(
+            TopoRouter.RD_OSPF6, os.path.join(CWD, "{}/ospf6d.conf".format(rname))
+        )
+
+    # Initialize all routers.
+    tgen.start_router()
+
+    # For debugging after starting FRR daemons, uncomment the next line
+    # tgen.mininet_cli()
+
+
+def teardown_module(mod):
+    "Teardown the pytest environment"
+    tgen = get_topogen()
+    tgen.stop_topology()
+
+def test_ospf6_converged():
+
+    tgen = get_topogen()
+
+    # Don't run this test if we have any failure.
+    if tgen.routers_have_failure():
+        pytest.skip(tgen.errors)
+
+    # For debugging, uncomment the next line
+    # tgen.mininet_cli()
+
+    # Wait for OSPF6 to converge  (All Neighbors in either Full or TwoWay State)
+    logger.info("Waiting for OSPF6 convergence")
+
+    # Set up for regex
+    pat1 = re.compile("^[0-9]")
+    pat2 = re.compile("Full")
+
+    timeout = 60
+    while timeout > 0:
+        logger.info("Timeout in %s: " % timeout),
+        sys.stdout.flush()
+
+        # Look for any node not yet converged
+        for router, rnode in tgen.routers().iteritems():
+            resStr = rnode.vtysh_cmd("show ipv6 ospf vrf {0}-cust1 neigh".format(router))
+
+            isConverged = False
+
+            for line in resStr.splitlines():
+                res1 = pat1.match(line)
+                if res1:
+                    isConverged = True
+                    res2 = pat2.search(line)
+
+                    if res2 == None:
+                        isConverged = False
```

Comment:
This is not appropriated, if there are two neighbors in the same router it will only check if the first converged.

Please try something similar to this:
https://github.com/FRRouting/frr/blob/master/tests/topotests/ospf6-topo1/test_ospf6_topo1.py#L209

Or check that all neighbors converged properly.

```py
def test_wait_protocol_convergence():
    "Wait for OSPFv3 to converge"
    tgen = get_topogen()
    if tgen.routers_have_failure():
        pytest.skip(tgen.errors)

    logger.info("waiting for protocols to converge")

    def expect_neighbor_full(router, neighbor):
        "Wait until OSPFv3 convergence."
        logger.info("waiting OSPFv3 router '{}'".format(router))
        test_func = partial(
            topotest.router_json_cmp,
            tgen.gears[router],
            "show ipv6 ospf6 neighbor json",
            {"neighbors": [{"neighborId": neighbor, "state": "Full"}]},
        )
        _, result = topotest.run_and_expect(test_func, None, count=130, wait=1)
        assertmsg = '"{}" convergence failure'.format(router)
        assert result is None, assertmsg

    expect_neighbor_full("r1", "10.0.0.2")
    expect_neighbor_full("r1", "10.0.0.3")

    expect_neighbor_full("r2", "10.0.0.1")
    expect_neighbor_full("r2", "10.0.0.3")

    expect_neighbor_full("r3", "10.0.0.1")
    expect_neighbor_full("r3", "10.0.0.2")
    expect_neighbor_full("r3", "10.0.0.4")

    expect_neighbor_full("r4", "10.0.0.3")
```